compiler/damlc/daml-stdlib-src/DA/Internal/Desugar.daml

```diff
@@ -46,9 +47,15 @@ data PreConsuming t = PreConsuming {}
 data Consuming t = Consuming {}
 data PostConsuming t = PostConsuming {}
 
--- Used to construct markers representing the fact that
--- a template `t` implements an interface `i`.
-data ImplementsT t i = ImplementsT
+-- Used to construct markers representing the fact that a
+-- (template or interface) `p` contains an interface instance of
+-- interface `i` for template `t`.
+-- `p` should equal `i` or `t`; this is checked during parsing
+-- and also during LF conversion.
+newtype InterfaceInstance p i t = InterfaceInstance ()
+
+mkInterfaceInstance : forall p i t. InterfaceInstance p i t
+mkInterfaceInstance = InterfaceInstance ()
```

Comment:
Instead of this we could just use the constructor `InterfaceInstance` directly, as we did before with `ImplementsT`, but going through a function would allow us to add constraints later on, which we could use for improved error messages.